### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/woodpecker/commit/1254f693d64d059649be605c29acc46288044848#commitcomment-89064200

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/woodpecker/commit/fda5bfc37797b58d3384dd1931bc15b7bdf097ee

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,7 +26,7 @@
 * SECURITY
   * Update github.com/containerd/containerd (#978) (#980)
 * BUGFIXES
-  * Return to page after clikcing login at navbar (#975) (#976)
+  * Return to page after clicking login at navbar (#975) (#976)
```
Comment (jsoref) [inline CHANGELOG.md:29]: Some projects don't like changing changelogs. Happy to drop this or anything else

File: CHANGELOG.md [hunk 2]
```diff
@@ -105,7 +105,7 @@
   * Add page to view all projects of a user / group (#741)
   * Let non required migration tasks fail and continue (#729)
   * Improve pipeline compiler (#699)
-  * Support ChangedFiles for Github & Gitlab PRs and pushes and Gitea pushes (#697)
+  * Support ChangedFiles for GitHub & GitLab PRs and pushes and Gitea pushes (#697)
```
Comment (jsoref) [inline CHANGELOG.md:108]: Brand.

File: cmd/server/server.go [hunk 3]
```diff
@@ -337,7 +337,7 @@ func (a *authorizer) streamInterceptor(srv interface{}, stream grpc.ServerStream
 	return handler(srv, stream)
 }
 
-func (a *authorizer) unaryIntercaptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+func (a *authorizer) unaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
```
Comment (jsoref) [inline cmd/server/server.go:340]: This seems likely to be private and thus harmless

File: docs/docs/30-administration/100-external-configuration-api.md [hunk 4]
```diff
@@ -84,7 +84,7 @@ WOODPECKER_CONFIG_SERVICE_ENDPOINT=https://example.com/ciconfig
   "configs": [
     {
       "name": ".woodpecker.yml",
-      "data": "pipeline:\n  backend:\n    image: alpine\n    commands:\n      - echo \"Hello there from Repo (.woodpecekr.yml)\"\n"
+      "data": "pipeline:\n  backend:\n    image: alpine\n    commands:\n      - echo \"Hello there from Repo (.woodpecker.yml)\"\n"
```
Comment (jsoref) [inline docs/docs/30-administration/100-external-configuration-api.md:87]: consistency...

File: docs/docs/91-migrations.md [hunk 5]
```diff
@@ -72,7 +72,7 @@ Some versions need some changes to the server configuration or the pipeline conf
 - Plugin Settings moved into `settings` section:
 
   ```diff
-   pipline:
+   pipeline:
```
Comment (jsoref) [inline docs/docs/91-migrations.md:75]: user facing...

File: server/cron/cron.go [hunk 6]
```diff
@@ -49,7 +49,7 @@ func Start(ctx context.Context, store store.Store, forge forge.Forge) error {
 
 				crons, err := store.CronListNextExecute(now.Unix(), checkItems)
 				if err != nil {
-					log.Error().Err(err).Int64("now", now.Unix()).Msg("obtain cron cron list")
+					log.Error().Err(err).Int64("now", now.Unix()).Msg("obtain cron list")
```
Comment (jsoref) [inline server/cron/cron.go:52]: 1. Does something care about this text?
2. I can't tell if this doubling is meaningful/intentional...

File: server/forge/bitbucket/internal/types.go [hunk 7]
```diff
@@ -158,7 +158,7 @@ type PullRequestHook struct {
 		Updated time.Time `json:"updated_on"`
 
 		Source struct {
-			Repo   Repo `json:"repsoitory"`
+			Repo   Repo `json:"repository"`
```
Comment (jsoref) [inline server/forge/bitbucket/internal/types.go:161]: scary

File: server/forge/bitbucket/internal/types.go [hunk 8]
```diff
@@ -169,7 +169,7 @@ type PullRequestHook struct {
 		} `json:"source"`
 
 		Dest struct {
-			Repo   Repo `json:"repsoitory"`
+			Repo   Repo `json:"repository"`
```
Comment (jsoref) [inline server/forge/bitbucket/internal/types.go:172]: scary

File: server/forge/coding/internal/webhook.go [hunk 9]
```diff
@@ -49,7 +49,7 @@ func (c *Client) AddWebhook(globalKey, projectName, link string) error {
 		u := fmt.Sprintf("/user/%s/project/%s/git/hook/%d", globalKey, projectName, webhook.ID)
 		params := url.Values{}
 		params.Set("hook_url", link)
-		params.Set("type_pust", "true")
+		params.Set("type_push", "true")
```
Comment (jsoref) [inline server/forge/coding/internal/webhook.go:52]: scary

File: server/forge/github/parse_test.go [hunk 10]
```diff
@@ -28,7 +28,7 @@ import (
 )
 
 const (
-	hookEvent  = "X-Github-Event"
+	hookEvent  = "X-GitHub-Event"
```
Comment (jsoref) [inline server/forge/github/parse_test.go:31]: I don't think that http headers are supposed to be case sensitive, so this change shouldn't matter, but it's notable.

File: server/forge/gitlab/gitlab.go [hunk 11]
```diff
@@ -46,14 +46,14 @@ const (
...
+	URL          string // GitLab server url.
 	ClientID     string // Oauth2 client id.
 	ClientSecret string // Oauth2 client secret.
 	SkipVerify   bool   // Skip ssl verification.
 }
 
-// Gitlab implements "Forge" interface
-type Gitlab struct {
+// GitLab implements "Forge" interface
+type GitLab struct {
```
Comment (jsoref) [inline server/forge/gitlab/gitlab.go:56]: This is potentially a public api and thus might need to be dropped...

File: server/store/datastore/migration/000_legacy_to_xorm.go [hunk 12]
```diff
@@ -96,7 +96,7 @@ var legacy2Xorm = task{
 				return err
 			}
 			if _, err := sess.Exec("INSERT INTO build_config (config_id, build_id) SELECT config_id,build_id FROM old_build_config;"); err != nil {
-				return fmt.Errorf("unable to set copy data in to temp table %s. Error: %v", "old_build_config", err)
+				return fmt.Errorf("unable to set copy data into temp table %s. Error: %v", "old_build_config", err)
```
Comment (jsoref) [inline server/store/datastore/migration/000_legacy_to_xorm.go:99]: This is my newest correction and has been the most controversial. Happy to drop. That said, given that the code itself uses `INTO`, I'm relatively confident in it.

File: server/store/store.go [hunk 13]
```diff
@@ -148,7 +148,7 @@ type Store interface {
 
 	// Logs
 	LogFind(*model.Step) (io.ReadCloser, error)
-	// TODO: since we do ReadAll in any case a ioReader is not the best idear
+	// TODO: since we do ReadAll in any case a ioReader is not the best idea
```
Comment (jsoref) [inline server/store/store.go:151]: Some repositories are cute, this one doesn't seem to be particularly playful.

File: web/components.d.ts [hunk 14]
```diff
@@ -45,7 +45,7 @@ declare module '@vue/runtime-core' {
     IMdiClockTimeEightOutline: typeof import('~icons/mdi/clock-time-eight-outline')['default']
     IMdiFormatListBulleted: typeof import('~icons/mdi/format-list-bulleted')['default']
     IMdiGestureTap: typeof import('~icons/mdi/gesture-tap')['default']
-    IMdiGithub: typeof import('~icons/mdi/github')['default']
+    IMdiGitHub: typeof import('~icons/mdi/github')['default']
```
Comment (jsoref) [inline web/components.d.ts:48]: This might be an API break...

**Review by jsoref — COMMENTED** [review]
Most corrections automatically suggested by Google Sheets. All fault mine.

File: web/components.d.ts (hunk 14 above)
Comment (lafriks) [reply web/components.d.ts:48]: this file is generated so can not be changed manually

Comment (jsoref) [reply web/components.d.ts]: Dropped

**Comment by jsoref** [comment]
@qwerty287: I'm guessing it doesn't like the rename of `Gitlab` to `GitLab`... I'll drop it at some point...

File: server/forge/bitbucket/internal/types.go (hunk 7 above)
Comment (anbraten) [reply server/forge/bitbucket/internal/types.go:161]: I don't think it was done on purpose. If we would have such a weird  case, I think we would add a comment.

**Comment by jsoref** [comment]
Yeah, I checked, there was only one instance I didn't change and it didn't seem remotely relevant. I've dropped the gitlab commit. I can revisit it later if the rest of this clears.

Fwiw, I've tried building locally and golang/modules hate me. Specifically some interaction between things and docker library versions (not the first time I've tripped on this, although I think in the past I've managed to get past it).

**Review by lafriks — APPROVED** [review]

File: cmd/server/server.go (hunk 3 above)
Comment (anbraten) [reply cmd/server/server.go:340]: Yes. Will be changed in #1189 again.

**Pull request merged**